pyiron_base/project/delayed.py

```diff
@@ -0,0 +1,223 @@
+import hashlib
+import inspect
+import re
+
+import cloudpickle
+
+
+def get_function_parameter_dict(funct):
+    return {
+        k: None if v.default == inspect._empty else v.default
+        for k, v in inspect.signature(funct).parameters.items()
+    }
+
+
+def get_hash(binary):
+    # Remove specification of jupyter kernel from hash to be deterministic
+    binary_no_ipykernel = re.sub(b"(?<=/ipykernel_)(.*)(?=/)", b"", binary)
+    return str(hashlib.md5(binary_no_ipykernel).hexdigest())
```

Comment:
I'm ok with this being here for now, but I think this should be moved to `pyiron_snippets` so that there's a unified hashing algorithm for all data.